src/components/GlobalPreferences/Notifications/SubscriptionsForm.js

```diff
@@ -0,0 +1,151 @@
+import React, { useState, useCallback } from 'react'
+import {
+  Box,
+  Button,
+  DropDown,
+  GU,
+  IconMail,
+  LoadingRing,
+  useTheme,
+} from '@aragon/ui'
+import { AppType } from '../../../prop-types'
+import PropTypes from 'prop-types'
+import memoize from 'lodash.memoize'
+import styled from 'styled-components'
+import { getEthNetworkType } from '../../../local-settings'
+import { createSubscription } from './notification-service-api'
+
+const getEventNamesFromAbi = memoize(abi =>
+  abi.filter(item => item.type === 'event').map(item => item.name)
+)
+export function SubscriptionsForm({ apps, dao, onApiError, onCreate, token }) {
+  // TODO: get subscriptions and filter out existing options
+  const [selectedAppIdx, setSelectedAppIdx] = useState(-1)
+  const [selectedEventIdx, setSelectedEventIdx] = useState(-1)
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const theme = useTheme()
+  const subscriptionApps = apps.filter(
+    ({ isAragonOsInternalApp }) => !isAragonOsInternalApp
+  )
+  const appNames = subscriptionApps.map(
+    app => `${app.name} ${app.identifier ? `(${app.identifier})` : ''}`
+  )
+  const selectedApp =
+    selectedAppIdx === -1 ? null : subscriptionApps[selectedAppIdx]
+  const eventNames =
+    selectedAppIdx === -1 ? [''] : getEventNamesFromAbi(selectedApp.abi)
+
+  const handleAppChange = index => {
+    setSelectedAppIdx(index)
+    setSelectedEventIdx(-1)
+  }
+  const handleEventChange = index => {
+    setSelectedEventIdx(index)
+  }
+  const handeSubscribe = async e => {
+    console.log(eventNames[selectedAppIdx])
+    setIsSubmitting(true)
+    const { abi, appName, proxyAddress } = selectedApp
+
+    try {
+      const payload = {
+        abi,
```

Comment:
We don't need to send the full abi; we can just send the short snippet that correlates to the event being subscribed, e.g.

```js
abi.find(({ name, type }) => type === 'event' && name === eventName)
```